lib/CL/devices/basic/basic.c

```diff
@@ -719,14 +719,21 @@ pocl_basic_fill_rect (void *data,
                 + buffer_slice_pitch * k, fill_pixel, pixel_size);
 }
 
+typedef unsigned long v2ul __attribute__ ((vector_size (16)));
```

Comment:
I mean not to say that you should measure and optimize this spot, I just have serious doubts that it makes sense to hand vectorize it here as it's not future proof and autovectorizers are surprisingly good these days.